### ISSUE TITLE
Not merging back to state with default-merge?

### ISSUE THREAD

**Issue by agzam** [anmonteiro/compassus#16]
I may be that Compassus not doing default-merge right? I wasn't seeing data coming from remote being merged back to the state. So I had to write a merge function like described in documentation:

https://github.com/omcljs/om/wiki/Documentation-(om.next)#reconciler-1

so this works:

```
(defn merge-fn 
  [reconciler state novelty query]
  {:next (merge state novelty)})
```

but if you do this, it doesn't:

```
(defn merge-fn 
  [reconciler state novelty query]
  {:keys (keys novelty)
   :next (merge state novelty)})
```

**Comment by anmonteiro** [comment]
Have you been able to repro this issue? I think we have tests for merging remote calls so I'm inclined to close this.

**Comment by agzam** [comment]
Check this out: https://github.com/agzam/trying-compassus/blob/master/src/cljs/sample_routing/core.cljs#L61
without `merge-fn` it does not work

**Comment by anmonteiro** [comment]
fixed d3a0b4523bcb3ffd6006ef6d46c4d0b6e53f028c
